### ISSUE TITLE
Decouple precalc from data:import

### ISSUE THREAD

**Issue by avmey** [seasketch/geoprocessing#196]
To support sub-regional planning, data will have to be imported at the largest planning level. keyStats need to be calculated for each subregion. 

Report developers should be able to:
- add/remove subregions within (add/remove islands)
- add/remove metrics
- add/remove dimensions (time, subregions)

Could have an editable regions.json file, and a `npm run precalc` call to get keyStats for each datasource-region pair.

**Comment by avmey** [comment]
A separate precalc is being tested in the Azores nearshore reports, subregional branch. The current format and workflows are:

Geographies stored in project/geographies.json in this format:
```
[
    {
        "geographyId": "nearshore",
        "datasourceId": "6nm_boundary",
        "display": "Nearshore"
    },
    {
        "geographyId": "santa-maria",
        "datasourceId": "santa-maria",
        "display": "Santa Maria"
    }
]
```

Keystats are generated using a `precalc.ts` script, which calculates keystats for each datasources-region pair. This output is stored in `precalc.json` to be used as denominators in reports. 

The current format of precalc.json is:
```
{
    "metrics": [
        {
            "metricId": "area",
            "value": 80000,
            "classId": "plateaus-total",
            "groupId": null,
            "geographyId": "nearshore",
            "sketchId": null
        },
        {
            "metricId": "area",
            "value": 5000,
            "classId": "plateaus-total",
            "groupId": null,
            "geographyId": "santa-maria",
            "sketchId": null
         },
    ]
}
```
(wherein `classId` is `datasourceId`+`-`+`class name` to allow for catagorical rasters and multiclass vectors)

The report components then access precalc.json for denominators to use in reports.

**Comment by twelch** [comment]
Resolved via #208
